### ISSUE TITLE
Does SimpleJson support tuples?

### ISSUE THREAD

**Issue by mvsmal** [Zaid-Ajaj/Fable.SimpleJson#13]
Hi, I faced an issue with `Fable.Remoting.Client` v4 (for Fable 2).
I have code of this kind (not perfect, buy I just want to simplify):
```fsharp
type Token = Token of string
type AuthenticationError =
    | TokenExpired
    | TokenInvalid
    | AccessDenied

type AlbumId = AlbumId of int
type AlbumAuthor = AlbumAuthor of string

type IMusicStore = {
  getAlbums : Token -> Async<Result<AlbumId * AlbumAuthor option, AuthenticationError>>
}
```

When I try to get the albums, function `getAlbums` never returns. In Network tab I see a successful response from the server, but it is never parsed by SimpleJson.
I tried to debug and reached this point:
https://github.com/Zaid-Ajaj/Fable.SimpleJson/blob/master/src/Json.Converter.fs#L140

Apparently a tuple `AlbumId * AlbumName option` is not supported by SimpleJson because in `Fable.Remoting.Client` v3, which is not using SimpleJson, it works well.

Am I missing something or there is a bug? I would love to help fixing it, just not sure how.
Thanks. 

UPDATE:
If I use record type instead of tuple, or even SingleCase DU, it works well

**Comment by Zaid-Ajaj** [comment]
Hello there @mvsmal, tuples should be supported as per the [tests](https://github.com/Zaid-Ajaj/Fable.SimpleJson/blob/master/test/Tests.fs#L738-L752), I will investigate this ASAP but the usual way to debugging and reproducing these issues is:
 - Add parser roundtrip tests (stringify and then parse)
 - Add parser tests from raw JSON strings (what you get from the server in the network tab)
 - Try combining complex values in parser tests

if these work, add a roundtrip test to Fable.Remoting.Client (to both ClientTests and ClientTestsV2) and run these in the integration tests: 
 - `./build.sh IntegrationTestsV2` (Fable 2.x)
 - `./build.sh IntegrationTests` (Fable 1.x) 

You can help me with adding a couple of tests to this repo and I will see to the rest :smile:

**Comment by mvsmal** [comment]
So I played around with SimpleJson and figured out that having this code:
```fsharp
type SingleCaseDU1 = SingleCaseDU1 of Guid
type SingleCaseDU2 = SingleCaseDU2 of int
type BasicDU = | Case1 | Case2

// I had to use NewGuid() in tests because Guid.Empty/Guid.Parse() fail. But it is another topic
let id = Guid.NewGuid()
let value : Result<SingleCaseDU1 * SingleCaseDU2 option, BasicDU> =
    (SingleCaseDU1 id, Some (SingleCaseDU2 1)) |> Ok

let str = Json.stringify value
```

`str` becomes: `["Ok",[["SingleCaseDU1","4358abc7-309f-4074-8b9f-838ada3ab5f8"],["SingleCaseDU2",1]]]`

But from server (`Fable.Remoting.Giraffe@3.4.0`) I receive 
`{"Ok":[{"SingleCaseDU1":"4358abc7-309f-4074-8b9f-838ada3ab5f8"},{"SingleCaseDU2":1}]}`

If I try to parse the string from server, I get this exception:
```
not ok 76 Json > Converting Result of tuple and DU
  ---
  message: "Died on test #1     at Module.<anonymous> (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:130137)
    at e (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:1:172)
    at Object.<anonymous> (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:1:63658)
    at e (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:1:172)
    at C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:1:964
    at Object.<anonymous> (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:1:975)
    at Module._compile (internal/modules/cjs/loader.js:689:30): Expected case 'Ok' to have 1 argument types but the JSON data only contained 2 values"
  severity: failed
  stack: Error: Expected case 'Ok' to have 1 argument types but the JSON data only contained 2 values
    at C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:42645
    at C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:42504
    at lu (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:93708)
    at hu (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:101185)
    at du (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:101390)
    at Object.<anonymous> (C:\Mine\Other\Proj\Fable.SimpleJson\public\bundle.js:8:130417)
    at runTest (C:\Mine\Other\Proj\Fable.SimpleJson\node_modules\qunitjs\qunit\qunit.js:1478:30)
    at Test.run (C:\Mine\Other\Proj\Fable.SimpleJson\node_modules\qunitjs\qunit\qunit.js:1464:6)
    at C:\Mine\Other\Proj\Fable.SimpleJson\node_modules\qunitjs\qunit\qunit.js:1670:12
    at Object.advance (C:\Mine\Other\Proj\Fable.SimpleJson\node_modules\qunitjs\qunit\qunit.js:1116:26)
```

I don't have any specific Json serializers, should I?

**Comment by Zaid-Ajaj** [comment]
@mvsmal Issue has been fixed, published both SimpleJson 2.5.0 and updated Fable.Remoting.Client to v4.3.0 it should be available soon from nuget after indexing

**Comment by mvsmal** [comment]
Thanks a lot!
